Harvest.Core/Services/SlothService.cs

```diff
@@ -0,0 +1,210 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Anlab.Core.Models.SlothModels;
+using Harvest.Core.Data;
+using Harvest.Core.Domain;
+using Harvest.Core.Models;
+using Harvest.Core.Models.Settings;
+using Harvest.Core.Models.SlothModels;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Serilog;
+
+namespace Harvest.Core.Services
+{
+    public interface ISlothService
+    {
+        Task<SlothResponseModel> MoveMoney(Transfer moneyTransfer);
+
+        Task ProcessTransferUpdates();
+    }
+
+    public class SlothService : ISlothService
+    {
+        private readonly AppDbContext _dbContext;
+        private readonly SlothSettings _slothSettings;
+        private readonly IFinancialService _financialService;
+
+        public SlothService(AppDbContext dbContext, IOptions<SlothSettings> slothSettings, IFinancialService financialService)
+        {
+            _dbContext = dbContext;
+            _slothSettings = slothSettings.Value;
+            _financialService = financialService;
+        }
+
+
+
+        //TODO: Add validation?
+        public async Task<SlothResponseModel> MoveMoney(Transfer moneyTransfer)
+        {
+            var token = _slothSettings.ApiKey;
+            var url = _slothSettings.ApiUrl;
+
+            if (string.IsNullOrWhiteSpace(token))
+            {
+                Log.Error("Sloth Token missing");
+            }
+
+            var debit = await _financialService.IsValid(moneyTransfer.FromAccount.Number);
+            var credit = await _financialService.IsValid(moneyTransfer.ToAccount.Number);
+
+            var model = new TransactionViewModel
+            {
+                MerchantTrackingNumber = moneyTransfer.Id.ToString(),
+                MerchantTrackingUrl = $"{_slothSettings.MerchantTrackingUrl}/{moneyTransfer.Id}"
+            };
+
+            model.Transfers.Add(new TransferViewModel
+            {
+                Account = moneyTransfer.FromAccount.Number?.ToUpperInvariant(),
+                Amount = moneyTransfer.Amount,
+                Chart = debit.KfsAccount.ChartOfAccountsCode, 
+                SubAccount = debit.KfsAccount.SubAccount,
+                Description = moneyTransfer.Description, 
```

Comment:
In other projects I've created a string extension that safely truncates these:

`        public static string SpecialTruncation(this string value, int additionalLength, int maxLength)
        {
            if ((value.Length + additionalLength) <= maxLength)
            {
                return value;
            }

            var max = (value.Length + additionalLength + 3) - maxLength;
            return $"{value.Substring(0, (value.Length - max))}...";
        }`

Reply:
Description = $"{order.Project.SpecialTruncation((order.RequestNum.Length + 3), 40)} - {order.RequestNum}",

Reply:
I found that function a little challenging to follow, so I split into two, `Truncate` and `TruncateAndAppend`